### PR TITLE
[Example] Low light image enhancement cc example with TF-lite

### PR DESCRIPTION
**Pull request by niley7464**
Low light image enhancement with Tenworflow-Lite model.
This example has a dependency on opencv4.

Signed-off-by: yelini-jeong <yelini.jeong@samsung.com>

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -96,6 +96,13 @@ if nnsquery_dep.found()
   have_nnsquery = true
 endif
 
+opencv_dep = dependency('opencv4', required: false)
+
+have_opencv = false
```
Comment (jaeyun-jung) [inline meson.build:101]: IMO, this is unnecessary.
nnstreamer_example_low_light_image_enhancement.py may not require opencv, 
then plz do not change native/meson.build and 
update native/example_low_light_image_enhancement/meson.build 
```
if opencv_dep.found()
  nnstreamer_example_low_light_image_enhancement = executable('nnstreamer_example_low_light_image_enhancement',
    'nnstreamer_example_low_light_image_enhancement.cc',
    dependencies: [glib_dep, gst_dep, gst_app_dep, opencv_dep],
    install: true,
    install_dir: examples_install_dir
  )
endif
```

File: native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  }
+
+  Mat img = imread(app->image_file);
+  if (!img.empty()) {
+    gint width = img.cols;
+    gint height = img.rows;
+    gint bpp = img.channels();
+    gint imagesize = width * height * bpp;
+
+    gint *buffer_array = new gint[imagesize];
```
Comment (jaeyun-jung) [inline native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc:165]: gint *buffer_array = (gint *) g_malloc(imagesize);
Use g_malloc() as gst_buffer_new_wrapped() releases allocated memory using g_free()

File: meson.build (hunk 1 above)
Comment (niley7464) [reply meson.build]: Thank you!
I updated it

File: native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    Mat enhancement_image = Mat(IMG_HEIGHT, IMG_WIDTH, CV_8UC3, image);
+    app = (app_data_s *)user_data;
+
+    gchar *enhancement_image_file =
+        g_strconcat("low_light_enhancement_", app->image_file, NULL);
+    imwrite(enhancement_image_file, enhancement_image);
+    g_print("%s created!\n", enhancement_image_file);
+    gst_memory_unmap(mem, &info);
+  }
+}
```
Comment (jaeyun-jung) [inline native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc:216]: plz add `gst_memory_unref(mem);` here.

gst_buffer_get_memory() increases ref count of memory. So it is necessary to decrease ref count after usage.

FYI:
If you don't want handling referred memory block, use gst_buffer_peek_memory() instead.
gst_buffer_peek_memory() gets memory block without increasing ref count in gst-buffer.

Comment (niley7464) [reply native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc:216]: I changed gst_buffer_get_memory() to gst_buffer_peek_memory()
I will keep in mind this next time!
Thank you

File: native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc (hunk 2 above)
Comment (niley7464) [reply native/example_low_light_image_enhancement/nnstreamer_example_low_light_image_enhancement.cc]: Thank you for the explanation
I changed it!

**Pull request merged**